src/backend/mi2/mi2.ts

```diff
@@ -886,3 +889,32 @@ export class MI2 extends EventEmitter implements IBackend {
 	protected stream;
 	protected sshConn;
 }
+function readEnvFile(filename: string): { [key: string]: string } {
```

Comment:
I would probably recommend taking these functions, along with the functionality in the MI2 constructor above and moving it to its own class (e.g., "Environment").  This would encapsulate this functionality and make it easy to unit test.

Reply:
Yes, that's a general issue with the existing code - the environment variables are _expected_ to be found in the inferior (this "just happens" when GDB starts it locally). If we cannot make this happen all environment settings should be dropped from all remote scenarios.
I think this should be changed in general to use; and I even forgot if there's a better way than `call setenv (name, value, 1)` [which likely only work for C/C++ code, not everything GDB supports].

Reply:
There does appear to be commands in both gdb and lldb for setting environment variables, so using that approach, rather than attempting to set them at the process level might be the way to go.  The environment variable setting commands could just be issued during the MI2::initCommands, similarly to how the path mappings are handled (i.e., extra commands).  The GDB documentation even says these environment variable settings are sent to gdbserver, so should support remote debugging as well (see [here](https://sourceware.org/gdb/onlinedocs/gdb/Environment.html#set-environment)).  This should also work with SSH connections too, since the commands are issued to the debugger itself.

- gdb Add/Remove Environment Variable:
```shell
(gdb) set environment FOO=BAR
(gdb) unset environment FOO
```

- lldb Add/Remove Environment Variable:
```shell
(lldb) settings set target.env-vars FOO=BAR
(lldb) settings remove target.env-vars FOO
```